activestorage/app/models/active_storage/blob/analyzable.rb

```diff
@@ -42,16 +42,21 @@ def analyzed?
     analyzed
   end
 
+  # Returns true if the blob's analyzer is present.
+  def analyze_later?
+    analyzer_class.present?
+  end
+
   private
     def extract_metadata_via_analyzer
       analyzer.metadata.merge(analyzed: true)
     end
 
     def analyzer
-      analyzer_class.new(self)
+      (analyzer_class || ActiveStorage::Analyzer::NullAnalyzer).new(self)
     end
 
     def analyzer_class
-      ActiveStorage.analyzers.detect { |klass| klass.accept?(self) } || ActiveStorage::Analyzer::NullAnalyzer
+      ActiveStorage.analyzers.detect { |klass| klass.accept?(self) }
```

Comment:
Think we should extract an `acceptable_analyzer_class` method instead of moving the NullAnalyzer into `analyzer`

`analyze_later?` would then be `acceptable_analyzer_class.present?`